src/uucore/src/lib/features/format/argument.rs

```diff
@@ -69,64 +73,87 @@ impl<'a> FormatArguments<'a> {
+            _ => "".as_ref(),
         }
     }
 
-    pub fn next_i64(&mut self, position: &ArgumentLocation) -> i64 {
+    pub fn next_i64(&mut self, position: &ArgumentLocation) -> Result<i64, FormatError> {
         match self.next_arg(position) {
-            Some(FormatArgument::SignedInt(n)) => *n,
-            Some(FormatArgument::Unparsed(s)) => extract_value(i64::extended_parse(s), s),
-            _ => 0,
+            Some(FormatArgument::SignedInt(n)) => Ok(*n),
+            Some(FormatArgument::Unparsed(os)) => Self::get_int::<i64>(os),
+            _ => Ok(0),
         }
     }
 
-    pub fn next_u64(&mut self, position: &ArgumentLocation) -> u64 {
+    pub fn next_u64(&mut self, position: &ArgumentLocation) -> Result<u64, FormatError> {
         match self.next_arg(position) {
-            Some(FormatArgument::UnsignedInt(n)) => *n,
-            Some(FormatArgument::Unparsed(s)) => {
-                // Check if the string is a character literal enclosed in quotes
-                if s.starts_with(['"', '\'']) {
-                    // Extract the content between the quotes safely using chars
-                    let mut chars = s.trim_matches(|c| c == '"' || c == '\'').chars();
-                    if let Some(first_char) = chars.next() {
-                        if chars.clone().count() > 0 {
-                            // Emit a warning if there are additional characters
-                            let remaining: String = chars.collect();
-                            show_warning!(
-                                "{}: character(s) following character constant have been ignored",
-                                remaining
-                            );
-                        }
-                        return first_char as u64; // Use only the first character
-                    }
-                    return 0; // Empty quotes
-                }
-                extract_value(u64::extended_parse(s), s)
-            }
-            _ => 0,
+            Some(FormatArgument::UnsignedInt(n)) => Ok(*n),
+            Some(FormatArgument::Unparsed(os)) => Self::get_int::<u64>(os),
+            _ => Ok(0),
         }
     }
 
-    pub fn next_extended_big_decimal(&mut self, position: &ArgumentLocation) -> ExtendedBigDecimal {
+    pub fn next_extended_big_decimal(
+        &mut self,
+        position: &ArgumentLocation,
+    ) -> Result<ExtendedBigDecimal, FormatError> {
         match self.next_arg(position) {
-            Some(FormatArgument::Float(n)) => n.clone(),
-            Some(FormatArgument::Unparsed(s)) => {
-                extract_value(ExtendedBigDecimal::extended_parse(s), s)
+            Some(FormatArgument::Float(n)) => Ok(n.clone()),
+            Some(FormatArgument::Unparsed(os)) => Self::get_int::<ExtendedBigDecimal>(os),
+            _ => Ok(ExtendedBigDecimal::zero()),
+        }
+    }
+
+    fn get_int<T>(os: &OsStr) -> Result<T, FormatError>
```

Comment:
whoops, note to self: change this to `get_num`, since `ExtendedBigDecimal` isn't an integer type.